### PR TITLE
fake rxInfo.Time if empty #2

### PR DESCRIPTION
**Pull request by comet-GmbH**
The Semtech packet-forwarder does not fill up the time and tmms fields if the gateway has no GPS.

Does nobody want to get a receive-time ?

This patch is for setting the time from system-time if it is empty.
A configuration flag is included to control behavior.
Should work on V2 + V3.

Something wrong ?

### CODE REVIEW

**Comment by comet-GmbH** [comment]
For accessing config.C package config needs to be imported.
This ends up in import cycle because config imports semtech.
Don't know how to prevent this at the moment.

**Comment by calvernaz** [comment]
Hm, it's unfortunate. I tested before and had no issues but maybe in a different package.

**Comment by comet-GmbH** [comment]
Found a solution.
It looks much better now.

File: internal/legacy/gateway/pf_structs.go [hunk 1]
```diff
@@ -379,13 +380,14 @@ func (d DatR) MarshalJSON() ([]byte, error) {
 func (d *DatR) UnmarshalJSON(data []byte) error {
 	i, err := strconv.ParseUint(string(data), 10, 32)
 	if err != nil {
-		d.LoRa = strings.Trim(string(data), `"`)
+		d.LoRa = strings.Trim(string(data), `"`) //"
```
Comment (calvernaz) [inline internal/legacy/gateway/pf_structs.go]: the comment looks strange, is well understood what the `Trim` function does.

Comment (comet-GmbH) [reply internal/legacy/gateway/pf_structs.go]: Just closing open " because it's confusing highlighter, etc...

**Pull request closed without merging**